src/main/js/component/common/ChannelTestModal.js

```diff
@@ -0,0 +1,72 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Modal} from 'react-bootstrap';
+import TextInput from "../../field/input/TextInput";
+
+class ChannelTestModal extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            destination: ''
+        };
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSendTestMessage = this.handleSendTestMessage.bind(this);
+    }
+
+    handleChange(event) {
+        event.preventDefault();
+        const target = event.target;
+        this.setState({destination: target.value});
+    }
+
+    handleSendTestMessage(event) {
+        event.preventDefault();
+        event.stopPropagation();
+        this.setState({
+            spinning: true
+        });
+        this.props.sendTestMessage(this.state.destination);
```

Comment:
Couldn't we set spinning to false again after the sendTestMessage() is complete?

Reply:
Yes, but the problem is that sendTestMessage does a dispatch asynchronously, so as soon as the method is called, the spinner is hidden. This results in the spinner not appearing at all.